src/main/kotlin/ee/carlrobert/codegpt/codecompletions/InfillRequestDetails.kt

```diff
@@ -1,55 +1,53 @@
 package ee.carlrobert.codegpt.codecompletions
 
-import com.intellij.codeInsight.inline.completion.InlineCompletionRequest
+import com.intellij.codeInsight.navigation.ImplementationSearcher
+import com.intellij.openapi.components.service
 import com.intellij.openapi.editor.Document
+import com.intellij.openapi.editor.Editor
 import com.intellij.openapi.util.TextRange
+import com.intellij.openapi.vfs.VirtualFile
 import ee.carlrobert.codegpt.EncodingManager
-import kotlin.math.max
-import kotlin.math.min
+import ee.carlrobert.codegpt.codecompletions.psi.CompletionContextService
 
-class InfillRequestDetails(val prefix: String, val suffix: String) {
+
+class InfillRequestDetails(val prefix: String, val suffix: String, val context: InfillContext?) :
+    ImplementationSearcher() {
     companion object {
         private const val MAX_OFFSET = 10_000
         private const val MAX_PROMPT_TOKENS = 128
 
-        fun fromInlineCompletionRequest(request: InlineCompletionRequest): InfillRequestDetails {
-            return fromDocumentWithMaxOffset(
-                request.editor.document,
-                request.editor.caretModel.offset,
-            )
-        }
-
-        private fun fromDocumentWithMaxOffset(
-            document: Document,
-            caretOffset: Int,
-        ): InfillRequestDetails {
-            val start = max(0, (caretOffset - MAX_OFFSET))
-            val end = min(document.textLength, (caretOffset + MAX_OFFSET))
-            return fromDocumentWithCustomRange(document, caretOffset, start, end)
-        }
-
-        private fun fromDocumentWithCustomRange(
-            document: Document,
+        fun fromInlineRequestDetails(
+            editor: Editor,
             caretOffset: Int,
-            start: Int,
-            end: Int,
+            prefix: String,
+            suffix: String
         ): InfillRequestDetails {
-            val prefix: String = truncateText(document, start, caretOffset, false)
-            val suffix: String = truncateText(document, caretOffset, end, true)
-            return InfillRequestDetails(prefix, suffix)
+            val truncatedPrefix = prefix.takeLast(MAX_OFFSET)
```

Comment:
We need to find a better solution for this `MAX_OFFSET` because 10,000 chars is a lot, especially since we're now providing more additional context.

Reply:
Yes definetely, perhaps we need to leverage the PSI tree to limit the completion to e.g. method or class level, depending on how big the file is. I remember I implemented the code completions in the beginning with limiting it to method-level instead of sending the entire file

Reply:
Yeah, I remember something similar. However, having just a method-level context isn't sufficient without the relevant context.